pkg/didcomm/common/didconnection/didconnection.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package didconnection
+
+import (
+	"encoding/json"
+	"fmt"
+
+	diddoc "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// BaseDIDConnectionStore stores DIDs indexed by key
```

Comment:
why the prefix Base?

Reply:
Another name would be welcome - `Store` is the interface name, and I didn't want a stuttering name. I can pull the interface out into an api package in the framework (which would allow this struct type to be named `Store`), though the interface is only necessary to allow mocking, and isn't intended to be implemented and injected by the client.

Reply:
If a package's only purpose is to implement a struct, it is okay to have that stutter.
I don't think `Base` adds any value.

(I'm also not understanding the conflict with the Store interface defined in another package?)

Reply:
The conflict would be with the interface in the same package - `didconnection.Store` in `didconnection/api.go`. But I've renamed to ConnectionStore so there's no conflict.